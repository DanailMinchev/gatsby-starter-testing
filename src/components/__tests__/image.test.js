import React from "react"
import { render } from "@testing-library/react"
import { useStaticQuery } from "gatsby" // mocked

import Image, { PureImage } from "../image"

describe(`Image`, () => {
  beforeEach(() => {
    useStaticQuery.mockImplementationOnce(() => ({
      placeholderImage: {
        childImageSharp: {
          fluid: {
            base64: "data:image/png;base64,pretend-i-am-a-base64-encoded-image",
            aspectRatio: 1,
            src: "/static/some-path/gatsby-astronaut.png",
            srcSet:
              "/static/some-path1/gatsby-astronaut.png 75w,\n/static/some-path2/gatsby-astronaut.png 150w,\n/static/some-path3/gatsby-astronaut.png 300w,\n/static/some-path4/gatsby-astronaut.png 450w,\n/static/some-path5/gatsby-astronaut.png 600w,\n/static/some-path6/gatsby-astronaut.png 800w",
            sizes: "(max-width: 300px) 100vw, 300px",
          },
        },
      },
    }))
  })

  it(`renders an image`, () => {
    const { container } = render(<Image />)

    expect(container.querySelector(`picture`)).toBeInTheDocument()
  })

  describe(`Snapshots`, () => {
    it(`matches snapshot`, () => {
      const { container } = render(<Image />)

      expect(container).toMatchSnapshot()
    })
  })
})

describe(`PureImage`, () => {
  const data = {
    placeholderImage: {
      childImageSharp: {
        fluid: {
          base64: "data:image/png;base64,pretend-i-am-a-base64-encoded-image",
          aspectRatio: 1,
          src: "/static/some-path/gatsby-astronaut.png",
          srcSet:
            "/static/some-path1/gatsby-astronaut.png 75w,\n/static/some-path2/gatsby-astronaut.png 150w,\n/static/some-path3/gatsby-astronaut.png 300w,\n/static/some-path4/gatsby-astronaut.png 450w,\n/static/some-path5/gatsby-astronaut.png 600w,\n/static/some-path6/gatsby-astronaut.png 800w",
          sizes: "(max-width: 300px) 100vw, 300px",
        },
      },
    },
  }

  it(`renders an image`, () => {
    const { container } = render(<PureImage data={data} />)

    expect(container.querySelector(`picture`)).toBeInTheDocument()
  })

  describe(`Snapshots`, () => {
    it(`matches snapshot`, () => {
      const { container } = render(<PureImage data={data} />)

      expect(container).toMatchSnapshot()
    })
  })
})
