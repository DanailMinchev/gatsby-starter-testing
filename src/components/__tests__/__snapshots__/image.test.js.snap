// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Snapshots matches snapshot 1`] = `
<div>
  <div
    class=" gatsby-image-wrapper"
    style="position: relative; overflow: hidden;"
  >
    <div
      aria-hidden="true"
      style="width: 100%; padding-bottom: 100%;"
    />
    <img
      alt=""
      aria-hidden="true"
      src="data:image/png;base64,pretend-i-am-a-base64-encoded-image"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
    />
    <picture>
      <source
        sizes="(max-width: 300px) 100vw, 300px"
        srcset="/static/some-path1/gatsby-astronaut.png 75w,
/static/some-path2/gatsby-astronaut.png 150w,
/static/some-path3/gatsby-astronaut.png 300w,
/static/some-path4/gatsby-astronaut.png 450w,
/static/some-path5/gatsby-astronaut.png 600w,
/static/some-path6/gatsby-astronaut.png 800w"
      />
      <img
        alt=""
        loading="lazy"
        sizes="(max-width: 300px) 100vw, 300px"
        src="/static/some-path/gatsby-astronaut.png"
        srcset="/static/some-path1/gatsby-astronaut.png 75w,
/static/some-path2/gatsby-astronaut.png 150w,
/static/some-path3/gatsby-astronaut.png 300w,
/static/some-path4/gatsby-astronaut.png 450w,
/static/some-path5/gatsby-astronaut.png 600w,
/static/some-path6/gatsby-astronaut.png 800w"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source srcset="/static/some-path1/gatsby-astronaut.png 75w,
/static/some-path2/gatsby-astronaut.png 150w,
/static/some-path3/gatsby-astronaut.png 300w,
/static/some-path4/gatsby-astronaut.png 450w,
/static/some-path5/gatsby-astronaut.png 600w,
/static/some-path6/gatsby-astronaut.png 800w" sizes="(max-width: 300px) 100vw, 300px" /&gt;&lt;img loading="lazy" sizes="(max-width: 300px) 100vw, 300px" srcset="/static/some-path1/gatsby-astronaut.png 75w,
/static/some-path2/gatsby-astronaut.png 150w,
/static/some-path3/gatsby-astronaut.png 300w,
/static/some-path4/gatsby-astronaut.png 450w,
/static/some-path5/gatsby-astronaut.png 600w,
/static/some-path6/gatsby-astronaut.png 800w" src="/static/some-path/gatsby-astronaut.png" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</div>
`;

exports[`PureImage Snapshots matches snapshot 1`] = `
<div>
  <div
    class=" gatsby-image-wrapper"
    style="position: relative; overflow: hidden;"
  >
    <div
      aria-hidden="true"
      style="width: 100%; padding-bottom: 100%;"
    />
    <img
      alt=""
      aria-hidden="true"
      src="data:image/png;base64,pretend-i-am-a-base64-encoded-image"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
    />
    <picture>
      <source
        sizes="(max-width: 300px) 100vw, 300px"
        srcset="/static/some-path1/gatsby-astronaut.png 75w,
/static/some-path2/gatsby-astronaut.png 150w,
/static/some-path3/gatsby-astronaut.png 300w,
/static/some-path4/gatsby-astronaut.png 450w,
/static/some-path5/gatsby-astronaut.png 600w,
/static/some-path6/gatsby-astronaut.png 800w"
      />
      <img
        alt=""
        loading="lazy"
        sizes="(max-width: 300px) 100vw, 300px"
        src="/static/some-path/gatsby-astronaut.png"
        srcset="/static/some-path1/gatsby-astronaut.png 75w,
/static/some-path2/gatsby-astronaut.png 150w,
/static/some-path3/gatsby-astronaut.png 300w,
/static/some-path4/gatsby-astronaut.png 450w,
/static/some-path5/gatsby-astronaut.png 600w,
/static/some-path6/gatsby-astronaut.png 800w"
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
      />
    </picture>
    <noscript>
      &lt;picture&gt;&lt;source srcset="/static/some-path1/gatsby-astronaut.png 75w,
/static/some-path2/gatsby-astronaut.png 150w,
/static/some-path3/gatsby-astronaut.png 300w,
/static/some-path4/gatsby-astronaut.png 450w,
/static/some-path5/gatsby-astronaut.png 600w,
/static/some-path6/gatsby-astronaut.png 800w" sizes="(max-width: 300px) 100vw, 300px" /&gt;&lt;img loading="lazy" sizes="(max-width: 300px) 100vw, 300px" srcset="/static/some-path1/gatsby-astronaut.png 75w,
/static/some-path2/gatsby-astronaut.png 150w,
/static/some-path3/gatsby-astronaut.png 300w,
/static/some-path4/gatsby-astronaut.png 450w,
/static/some-path5/gatsby-astronaut.png 600w,
/static/some-path6/gatsby-astronaut.png 800w" src="/static/some-path/gatsby-astronaut.png" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
  </div>
</div>
`;
